documentation/src/main/asciidoc/user_guide/execute_grid.adoc

```diff
@@ -0,0 +1,163 @@
+the command again.
+When this occurs the Cluster Executor will choose a single node again to resubmit the command to
+up to the desired number of failover attempts. Note the chosen node could be any node that passes
+the topology or predicate check. Failover is enabled by invoking the overridden
+link:{javadocroot}/org/infinispan/manager/ClusterExecutor.html#singleNodeSubmission-int-[singleNodeSubmission]
+method. The given command will be resubmitted again to a single node until either
+the command completes without exception or the total submission amount is equal to the provided
+failover count. If however a `org.infinispan.util.concurrent.TimeoutException` is encountered
+either from the command or by the command timing out as defined in `timeout` then it will
+terminate the failover process and complete any pending CompletableFutures with the
+exception.
+
+
+==== Example: Dynamically Start and Stop Clustered Cache
+This example shows how you can use the ClusterExecutor to dynamically start and stop a cache.
+
+===== Non-Clustered
+
+Start start/stop cache in non-clustered mode is simple.  You can use _EmbeddedCacheManager.defineConfiguration(cacheName, configuration)_ to define a cache, and then call _EmbeddedCacheManager.getCache(cacheName)_.
+
+If you don't define a specific configuration for the cache and directly call _EmbeddedCacheManager.getCache(...)_ , then a new cache would be created with default configurations.
+
+To stop a cache, call _EmbeddedCacheManager.remove(cacheName)_
+
+===== Clustered
+To start a clustered cache, you'll need to do the above on every clustered node, while making sure the cache mode is clustered, of course.
+
+You can start the cache by calling _EmbeddedCacheManager.getCache(...)_
+To do this on every single node though, you could write your own service to do that, or with JMX, or use the ClusterExecutor.
+
+[source,java]
+.StartCache.java
+----
+
+   EmbeddedCacheManager manager = ...;
+   String cacheName = "start-this-cache";
+   manager.executor().submitConsumer(localManager -> {
+         localManager.getCache(cacheName);
+         return null;
+      }, (address, value, throwable) -> {
+      if (throwable != null) {
+         log.fatal("Cache startup encountered exception on node " + address, t);
+      }
+   }).join();
+
+----
+
+The first argument is a `Function` that when invoked will pass the `EmbeddedCacheManager` local to each
+node.  Normally this also allows for a return value to be sent back, but unfortunately a `Cache`
+instance is not serializable so we can't send that back to the calling node.  Thus we have to
+return null.
+In this case the second argument `TriConsumer` would be called back for each node and will contain
+who this response is from (address), the return value (if there was one, in our case this is always
+null), and a throwable if a problem occurred.  The value and throwable variables will never both
```

Comment:
Got it, yeah I am fine with this. Maybe I should add this override and deprecate the other (to be removed in 10.0) ?